che-theia-terminal/src/browser/terminal-widget/remote-terminal-widget.ts

```diff
@@ -0,0 +1,389 @@
+        this.closeOnDispose = false;
+        return { terminalId: this.terminalId, titleLabel: this.title.label };
+    }
+
+    restoreState(oldState: object) {
+        if (this.restored === false) {
+            const state = oldState as { terminalId: number, titleLabel: string };
+            /* This is a workaround to issue #879 */
+            this.restored = true;
+            this.title.label = state.titleLabel;
+            this.start(state.terminalId);
+        }
+    }
+
+    /* Get the font family and size from the CSS custom properties defined in
+       the root element.  */
+    private getCSSPropertiesFromPage(): TerminalCSSProperties {
+        /* Helper to look up a CSS property value and throw an error if it's
+           not defined.  */
+        function lookup(props: CSSStyleDeclaration, name: string): string {
+            /* There is sometimes an extra space in the front, remove it.  */
+            const value = htmlElementProps.getPropertyValue(name).trim();
+            if (!value) {
+                throw new Error(`Couldn\'t find value of ${name}`);
+            }
+
+            return value;
+        }
+
+        /* Get the CSS properties of <html> (aka :root in css).  */
+        const htmlElementProps = getComputedStyle(document.documentElement);
+
+        const fontFamily = lookup(htmlElementProps, '--theia-code-font-family');
+        const fontSizeStr = lookup(htmlElementProps, '--theia-code-font-size');
+        const foreground = lookup(htmlElementProps, '--theia-ui-font-color1');
+        const background = lookup(htmlElementProps, '--theia-layout-color3');
+
+        /* The font size is returned as a string, such as ' 13px').  We want to
+           return just the number of px.  */
+        const fontSizeMatch = fontSizeStr.trim().match(/^(\d+)px$/);
+        if (!fontSizeMatch) {
+            throw new Error(`Unexpected format for --theia-code-font-size (${fontSizeStr})`);
+        }
+
+        const fontSize = Number.parseInt(fontSizeMatch[1]);
+
+        /* xterm.js expects #XXX of #XXXXXX for colors.  */
+        const colorRe = /^#([0-9a-fA-F]{3}|[0-9a-fA-F]{6})$/;
+
+        if (!foreground.match(colorRe)) {
+            throw new Error(`Unexpected format for --theia-ui-font-color1 (${foreground})`);
+        }
+
+        if (!background.match(colorRe)) {
+            throw new Error(`Unexpected format for --theia-layout-color3 (${background})`);
+        }
+
+        return {
+            fontSize: fontSize,
+            fontFamily: fontFamily,
+            foreground: foreground,
+            background: background,
+        };
+    }
+
+    protected registerResize(): void {
+        this.term.on('resize', size => {
+            if (this.terminalId === undefined) {
+                return;
+            }
+
+            if (!size) {
+                return;
+            }
+
+            this.cols = size.cols;
+            this.rows = size.rows;
+
+            let resizeParam: ResizeParam = {
```

Comment:
A lot of variables within the PR are declared as `let` but they're actually constants. Should they be declared as `const`?

Reply:
Yes, it should be.